ehsm_kms_service/secret_manager_apis.js

```diff
@@ -0,0 +1,279 @@
+            return
+        }
+        if (!checkStringParam(versionId, true)) {
+            res.send(_result(400, 'versionId cannot be empty and must be string'))
+            return
+        }
+        if (!checkStringParam(description, false)) {
+            res.send(_result(400, 'description must be string'))
+            return
+        }
+        if (!checkStringParam(rotationInterval, false)) {
+            res.send(_result(400, 'rotationInterval must be string'))
+            return
+        }
+        if (!checkStringParam(tags, false)) {
+            res.send(_result(400, 'tags must be string'))
+            return
+        }
+        if (!((/^([a-z\d]{8}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{12})$/).test(encryptionKeyId))) {
+            res.send(_result(400, 'encryptionKeyId format wrong'))
+            return
+        }
+        if (tags != '' && tags != undefined) {
+            tags_jsonArray = buildTagsArray(tags)
+            if (!tags_jsonArray) {
+                res.send(_result(400, 'tags format error'))
+                return
+            }
+        }
+
+        const secret_name_query = {
+            selector: {
+                appid,
+                secretName
+            },
+            fields: ['appid', 'secretName'],
+            limit: 1,
+        }
+        let secret_name_result = await DB.partitionedFind('secret_metadata', secret_name_query)
+        if (secret_name_result.docs.length > 0) {
+            res.send(_result(400, 'Secret name already exists, cannot use the same secret name'))
+            return
+        }
+
+        if (encryptionKeyId != '' && typeof (encryptionKeyId) == 'string') {
+            let key_id_res = await DB.partitionedFind('cmk', queryKeyId(appid, encryptionKeyId))
+            if (key_id_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find key id'))
+                return
+            } else {
+                keyBlob = key_id_res.docs[0].keyBlob
+            }
+        } else if (encryptionKeyId == "" || encryptionKeyId == undefined) {
+            let default_cmk_res = await DB.partitionedFind('user_info', queryDefaultCMK(appid))
+            if (default_cmk_res.docs.length == 0) {
+                res.send(_result(400, 'Cannot find default CMK'))
+                return
+            }
+            keyBlob = default_cmk_res.docs[0].sm_default_cmk
+        } else {
+            res.send(_result(400, 'encryptionKeyId error'))
+            return
+        }
+        const apikey_encrypt_res = napi_result(cryptographic_apis.Encrypt, res, [keyBlob, secretData, ''])
+        const { ciphertext } = apikey_encrypt_res.result
+
+        DB.insert({
+            _id: `secret_metadata:${uuidv4()}`,
+            appid,
+            secretName,
+            encryptionKeyId,
+            description,
+            createTime,
+            deleteTime: null,
+            plannedDeleteTime: null,
+            rotationInterval,
```

Comment:
rotation support will be in another patch, right?

Reply:
Yes, now only one field is reserved